### PR TITLE
8330016: Stress seed should be initialized for runtime stub compilation

### PR DESCRIPTION
**Pull request by danielogh**
We can initialize the stress seed for runtime stub compilation as we already do for method compilation. This found the bug described in JDK-8329258. It would apply if StressGCM or StressLCM vm flags are set.

Testing: T1-5 default options. T1-5 with -XX:+StressLCM and -XX:+StressGCM. Manually tested that the stress seed is set and printed to compilation log if either stress option is set.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330016](https://bugs.openjdk.org/browse/JDK-8330016): Stress seed should be initialized for runtime stub compilation (**Enhancement** - P4)

### Reviewers
 * [Roberto Castañeda Lozano](https://openjdk.org/census#rcastanedalo) (@robcasloz - **Reviewer**)
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [6ccef597](https://git.openjdk.org/jdk/pull/19095/files/6ccef5975b37a89406430c9320517d995320d61f)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19095/head:pull/19095` \
`$ git checkout pull/19095`

Update a local copy of the PR: \
`$ git checkout pull/19095` \
`$ git pull https://git.openjdk.org/jdk.git pull/19095/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19095`

View PR using the GUI difftool: \
`$ git pr show -t 19095`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19095.diff">https://git.openjdk.org/jdk/pull/19095.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19095#issuecomment-2095286341)

### CODE REVIEW

File: src/hotspot/share/opto/compile.cpp [hunk 1]
```diff
@@ -5068,6 +5063,18 @@ void Compile::remove_speculative_types(PhaseIterGVN &igvn) {
 
 // Auxiliary methods to support randomized stressing/fuzzing.
 
+void Compile::initialize_stress_seed(DirectiveSet* directive) {
```
Comment (robcasloz) [inline src/hotspot/share/opto/compile.cpp:5066]:
```suggestion
void Compile::initialize_stress_seed(const DirectiveSet* directive) {
```

File: src/hotspot/share/opto/compile.hpp [hunk 2]
```diff
@@ -1278,6 +1278,9 @@ class Compile : public Phase {
   int random();
   bool randomized_select(int count);
 
+  // seed random number generation and log the seed for repeatability.
+  void initialize_stress_seed(DirectiveSet* directive);
```
Comment (robcasloz) [inline src/hotspot/share/opto/compile.hpp:1282]:
```suggestion
  void initialize_stress_seed(const DirectiveSet* directive);
```

**Review by robcasloz — APPROVED** [review]
Looks good otherwise!

**Review by chhagedorn — APPROVED** [review]
Looks good to me, too.

**Review by robcasloz — APPROVED** [review]

**Comment by danielogh** [comment]
Thanks for the reviews!

**Pull request closed without merging**